src/config/backends/odmdata.ts

```diff
@@ -0,0 +1,242 @@
+				'Unable to parse ODMDATA configuration. Data at offsets do not match.',
+			);
+			throw new ODMDataError(
+				'Unable to parse ODMDATA configuration. Data at offsets do not match.',
+			);
+		}
+		// Find the configuration given the optionsBuffer
+		const configIndex = this.CONFIG_BYTES.reduce(
+			(currentIndex: number, _config: number, index: number) => {
+				if (
+					this.CONFIG_BUFFER.readUInt8(index) === optionsBuffer.readUInt8(0)
+				) {
+					return index;
+				}
+				return currentIndex;
+			},
+			null,
+		);
+		// Check if we found a configuration we support
+		if (configIndex === null) {
+			log.error(
+				`ODMDATA is set with an unsupported byte: 0x${optionsBuffer.readUInt8(
+					0,
+				)}`,
+			);
+			throw new ODMDataError(
+				'Unable to parse ODMDATA configuration. Unsupported configuration byte set',
+			);
+		}
+		// Return supported configuration number currently set
+		return {
+			configuration: `${configIndex + 1}`,
+		};
+	}
+
+	private async setByteAtOffset(byte: number, offset: number): Promise<void> {
+		// Obtain a file handle on bootConfigPath
+		const fileHandle = await this.getFileHandle(Odmdata.bootConfigPath);
+		try {
+			await fs.write(fileHandle, byte, offset);
+		} catch (e) {
+			log.error(`Issue writing to '${Odmdata.bootConfigPath}'`);
+			throw e;
+		} finally {
+			await fileHandle.close();
+		}
+	}
+
+	private async getFileHandle(
+		file: string,
+		flags = 'r+', // Open file for reading and writing. An exception occurs if the file does not exist.
+	): Promise<fs.FileHandle> {
+		try {
+			return await fs.open(file, flags);
+		} catch (e) {
+			switch (e.code) {
+				case 'ENOENT':
+					log.error(`File not found at: ${Odmdata.bootConfigPath}`);
+					throw new ODMDataError(
+						`File not found at: ${Odmdata.bootConfigPath}`,
+					);
+				case 'EACCES':
+					log.error(
+						`Permission denied when opening '${Odmdata.bootConfigPath}'`,
+					);
+					throw new ODMDataError(
+						`Permission denied when opening '${Odmdata.bootConfigPath}'`,
+					);
+				default:
+					log.error(`Unknown error when opening '${Odmdata.bootConfigPath}'`);
+					throw new ODMDataError(
+						`Unknown error when opening '${Odmdata.bootConfigPath}'`,
+					);
+			}
+		}
+	}
+
+	private async readBootConfigPath(): Promise<Buffer> {
+		// Create a buffer to store config byte values
+		const valuesBuffer = Buffer.alloc(3);
+		// Obtain a file handle on bootConfigPath
+		const fileHandle = await this.getFileHandle(Odmdata.bootConfigPath);
+		// Set single byte values in buffer at ODMDATA offsets
+		try {
+			for (let offset = 0; offset < 3; offset++) {
+				await fs.read(
+					fileHandle,
+					valuesBuffer,
+					offset,
+					1,
+					this.BYTE_OFFSETS[offset],
+				);
+			}
+		} catch (e) {
+			log.error(`Issue reading '${Odmdata.bootConfigPath}'`);
+			throw e;
+		} finally {
+			await fileHandle.close();
+		}
+		return valuesBuffer;
+	}
+
+	private async setReadOnly(value: boolean): Promise<void> {
```

Comment:
Please name this parameter more accurately.

Reply:
I think this is acceptable because the name of the function is simple and clear on what the single parameter is for. If there were more parameters passed I would change `value` to something more clear. Thoughts?